src/object/cli_obj.c

```diff
@@ -4190,18 +4190,22 @@ obj_list_get_shard(struct obj_auxi_args *obj_auxi, unsigned int map_ver,
 		int leader;
 
 		leader = obj_grp_leader_get(obj, shard, map_ver);
+		if (leader < 0) {
+			/* return nonapplicable in case the caller
+			 * want to retry non-leader option.
+			 */
+			D_DEBUG(DB_IO, DF_OID" failed to leader %d\n",
+				DP_OID(obj->cob_md.omd_id), leader);
+			obj_auxi->to_leader = 0;
+			D_GOTO(out, shard = -DER_NOTAPPLICABLE);
```

Comment:
maybe can just D_GOTO(out, shard = leader)?

Reply:
oh, we need specific failure to retry on non-leader shard in this case. see migrate_one_epoch_object(), so it has to use DER_NOTAPPLICABLE here.